src/blockdir.rs

```diff
@@ -242,89 +240,68 @@ impl BlockDir {
     }
 
     /// Return all the blocknames in the blockdir, in arbitrary order.
+    // TODO: Add a monitor since this operation might takes longer.
     pub fn block_names(&self) -> Result<impl Iterator<Item = BlockHash>> {
-        let progress = nutmeg::View::new("List blocks", ui::nutmeg_options());
-        progress.update(|_| ());
         Ok(self
             .iter_block_dir_entries()?
             .filter_map(|de| de.name.parse().ok()))
     }
 
     /// Return all the blocknames in the blockdir.
-    pub fn block_names_set(&self) -> Result<HashSet<BlockHash>> {
-        let progress = nutmeg::View::new(UnboundedModel::new("List blocks"), ui::nutmeg_options());
-        Ok(self
+    pub fn block_names_set(&self, monitor: &dyn ValidateMonitor) -> Result<HashSet<BlockHash>> {
+        let mut block_count = 0usize;
+
+        monitor.list_block_names(block_count);
```

Comment:
The name of this function reads a bit strangely here because we're not asking the monitor to list the block names. Maybe `estimated_block_count` or `blocks_counted`.

Reply:
mm, maybe not worth changing now.